parser/parser.go

```diff
@@ -0,0 +1,221 @@
+// go run parser.go [file_path] [semster]
+// example: go run parser.go "Fall 2019.csv" 19F
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"parser/configs"
+	"parser/model"
+)
+
+type Class struct {
+	subject           string
+	catalogNumber     string
+	section           string
+	gradeDistribution bson.A
+}
+
+func getCollection(client *mongo.Client, database string, collection string) (returnCollection *mongo.Collection) {
+	return (client.Database(database).Collection(collection))
+}
+
+func DBConnect(URI string) (client *mongo.Client) {
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(URI))
+	if err != nil {
+		fmt.Println(err)
+		panic(err)
+	}
+	if err := client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		fmt.Println(err)
+		panic(err)
+	}
+	return client
+}
+
+func csvToClassesSlice(csvFile *os.File, logFile *os.File) (classes []Class) {
+	reader := csv.NewReader(csvFile)
+	records, err := reader.ReadAll() // records is [][]strings
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	// look for the subject column and w column
+	subjectCol := -1
+	wCol := -1
+	for j := 0; j < len(records[0]); j++ {
+		if strings.Compare(records[0][j], "Subject") == 0 {
+			subjectCol = j
+		}
+		if strings.Compare(records[0][j], "W") == 0 || strings.Compare(records[0][j], "Total W") == 0 || strings.Compare(records[0][j], "W Total") == 0 {
+			wCol = j
+		}
+		if wCol == -1 || subjectCol == -1 {
+			continue
+		} else {
+			break
+		}
+	}
+	if wCol == -1 {
+		logFile.WriteString("could not find W column")
+	}
+	catalogNumberCol := subjectCol + 1
```

Comment:
check to make sure these fall within the bounds of the records 2d slice

Reply:
These are now set by iterating through the for loop, and would be in bounds.